scripts/deploy-federation-with-helm-chart.sh

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script automates deployment of a federation - as documented in
+# the README - to the current kubectl context.  It also joins the
+# hosting cluster as a member of the federation.
+#
+# WARNING: The service account for the federation namespace will be
+# granted the cluster-admin role.  Until more restrictive permissions
+# are used, access to the federation namespace should be restricted to
+# trusted users.
+#
+# If using minikube, a cluster must be started prior to invoking this
+# script:
+#
+#   $ minikube start
+#
+# This script depends on helm and kubectl being installed in
+# the path.  These and other test binaries can be installed via the
+# download-binaries.sh script, which downloads to ./bin:
+#
+#   $ ./scripts/download-binaries.sh
+#   $ export PATH=$(pwd)/bin:${PATH}
+#
+# To redeploy federation from scratch, prefix the deploy invocation with the deletion script:
+#
+#   # WARNING: The deletion script will remove federation and cluster registry data
+#   $ ./scripts/delete-federation-with-helm-chart.sh [join-cluster]... && ./scripts/deploy-federation-with-helm-chart.sh <image> [join-cluster]...
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+
+NS="${FEDERATION_NAMESPACE:-federation-system}"
+PUBLIC_NS=kube-multicluster-public
+IMAGE_NAME="${1:-}"
+NAMESPACED="${NAMESPACED:-}"
+SYNC_CHART="${SYNC_CHART:-}"
+
+LATEST_IMAGE_NAME=quay.io/kubernetes-multicluster/federation-v2:latest
+USE_LATEST=
+if [[ "${IMAGE_NAME}" == "$LATEST_IMAGE_NAME" ]]; then
+  USE_LATEST=y
+fi
+
+KF_NS_ARGS="--federation-namespace=${NS} "
+if [[ "${NAMESPACED}" ]]; then
+  KF_NS_ARGS+="--registry-namespace=${NS} --limited-scope=true"
```

Comment:
As above, it is not request `registry-namespace` for `NAMESPACED` federation.

Reply:
as above